### PR TITLE
OPHJOD-137

### PR DESCRIPTION
**Pull request by jhyoty**
- Add Docker container for the backend application
- Create build and deploy pipeline (OPHJOD-134) using GitHub actions
(See also:  https://github.com/Opetushallitus/jod-yksilo-infra/pull/5)

**JIRA ticket** 
https://jira.eduuni.fi/browse/OPHJOD-137

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -1,33 +1,130 @@
...
+      - uses: actions/checkout@v4
+
+      - uses: actions/download-artifact@v4
+        with:
+          name: application
+          path: build/libs
+
+      - uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: ${{ vars.JOD_UTIL_AWS_ROLE_ARN }}
```
Comment (sauanto) [inline .github/workflows/build.yml:59]: The role could be divided as follows:
```
role-to-assume: arn:aws:iam::${{ secrets.JOD_AWS_ACCOUNT_ID }}:role/${{ secrets.JOD_UTIL_AWS_ROLE_NAME}}
```

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -1,33 +1,130 @@
...
+    if: github.ref == 'refs/heads/main' || github.event_name == 'workflow_dispatch'
+    permissions:
+      id-token: write
+    needs: package
+    runs-on: ubuntu-latest
+    environment: dev
+    steps:
+      - uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: ${{ vars.JOD_AWS_DEPLOY_ROLE_ARN }}
```
Comment (sauanto) [inline .github/workflows/build.yml:108]: The role could be divided.

File: src/main/resources/application.yml [hunk 3]
```diff
@@ -7,16 +7,11 @@ spring:
...
-  error:
-    include-stacktrace: never
-    include-exception: false
-
 springdoc:
   api-docs:
     enabled: true
     path: /openapi/openapi.json
-  packages-to-scan: "org.example"
+  packages-to-scan: "fi.okm.jod"
```
Comment (sauanto) [inline src/main/resources/application.yml:14]: Maybe?
```
packages-to-scan: "fi.okm.jod.yksilo"
```

**Review by sauanto — APPROVED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (jhyoty) [reply .github/workflows/build.yml:59]: Util role has a different account from the deploy role, so splitting would make configuration more complex, keeping as it is.

**Pull request merged**